src/charm.py

```diff
@@ -35,83 +33,73 @@ def __init__(self, *args: typing.Any):
             args: Arguments passed to the CharmBase parent constructor.
         """
         super().__init__(*args)
-        self.framework.observe(self.on.httpbin_pebble_ready, self._on_httpbin_pebble_ready)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
-
-    def _on_httpbin_pebble_ready(self, event: ops.PebbleReadyEvent) -> None:
-        """Define and start a workload using the Pebble API.
-
-        Change this example to suit your needs. You'll need to specify the right entrypoint and
-        environment configuration for your specific workload.
-
-        Learn more about interacting with Pebble at at https://juju.is/docs/sdk/pebble.
-
-        Args:
-            event: event triggering the handler.
+        self.certificates = CertificatesObserver(self)
+        self.traefik_route = TraefikRouteObserver(self)
+        self.opensearch = OpenSearchRequires(self, OPENSEARCH_RELATION_NAME, "placeholder")
+        try:
+            opensearch_relation = self.model.get_relation(OPENSEARCH_RELATION_NAME)
+            opensearch_relation_data = (
+                opensearch_relation.data[opensearch_relation.app] if opensearch_relation else {}
+            )
+            self.state = State.from_charm(self, opensearch_relation_data)
+        except InvalidStateError:
+            self.unit.status = ops.BlockedStatus()
+            return
+        self.framework.observe(
+            self.on.opensearch_client_relation_changed, self._on_opensearch_client_relation_changed
+        )
+        self.framework.observe(
+            self.on.wazuh_server_pebble_ready, self._on_wazuh_server_pebble_ready
+        )
+
+    def _on_wazuh_server_pebble_ready(self, _: ops.PebbleReadyEvent) -> None:
+        """Pebble ready handler for the wazuh-server container."""
+        self._reconcile()
+
+    def _on_opensearch_client_relation_changed(self, _: ops.RelationJoinedEvent) -> None:
+        """Pebble ready handler for the wazuh-indexer relation changed event."""
+        self._reconcile()
+
+    def _reconcile(self) -> None:
+        """Reconcile Wazuh configuration with charm state.
+
+        This is the main entry for changes that require a restart.
         """
-        # Get a reference the container attribute on the PebbleReadyEvent
-        container = event.workload
-        # Add initial Pebble config layer using the Pebble API
-        container.add_layer("httpbin", self._pebble_layer, combine=True)
-        # Make Pebble reevaluate its plan, ensuring any services are started if enabled.
+        container = self.unit.get_container("wazuh-server")
+        if not container.can_connect():
+            logger.warning(
+                "Unable to connect to container during reconcile. "
+                "Waiting for future events which will trigger another reconcile."
```

Comment:
I'm concerned that this might prevent the charm from going into an error state, leading to a false active status. WDYT